src/ort_genai_c.h

```diff
@@ -137,13 +137,8 @@ OGA_EXPORT OgaResult* OGA_API_CALL OgaCreateGeneratorParams(const OgaModel* mode
  */
 OGA_EXPORT void OGA_API_CALL OgaDestroyGeneratorParams(OgaGeneratorParams* generator_params);
 
-/*
- * \brief Sets the maximum length that the generated sequence can have.
- * \param[in] params The generator params to set the maximum length on.
- * \param[in] max_length The maximum length of the generated sequences.
- * \return OgaResult containing the error message if the setting of the maximum length failed.
- */
-OGA_EXPORT OgaResult* OGA_API_CALL OgaGeneratorParamsSetMaxLength(OgaGeneratorParams* generator_params, size_t max_length);
+OGA_EXPORT OgaResult* OGA_API_CALL OgaGeneratorParamsSetSearchNumber(OgaGeneratorParams* generator_params, const char* name, double value);
+OGA_EXPORT OgaResult* OGA_API_CALL OgaGeneratorParamsSetSearchBool(OgaGeneratorParams* generator_params, const char* name, bool value);
```

Comment:
how about 2 overload functions with name OgaGeneratorParamsSetSearchOptions?

Reply:
Just like what C# APIs

Reply:
Never mind. C doesn't support overload

Reply:
I would, but C can't do overloaded functions. No parameter name mangling in C.